internal/crowdstrike/data_falcon_detection_details.go

```diff
@@ -0,0 +1,120 @@
+package crowdstrike
+
+import (
+	"context"
+
+	"github.com/crowdstrike/gofalcon/falcon"
+	"github.com/crowdstrike/gofalcon/falcon/client/detects"
+	"github.com/crowdstrike/gofalcon/falcon/models"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin"
+	"github.com/blackstork-io/fabric/plugin/dataspec"
+	"github.com/blackstork-io/fabric/plugin/dataspec/constraint"
+	"github.com/blackstork-io/fabric/plugin/plugindata"
+)
+
+func makeFalconDetectionDetailsDataSource(loader ClientLoaderFn) *plugin.DataSource {
+	return &plugin.DataSource{
+		Doc:      "The `falcon_detection_details` data source fetches detection details from Falcon API.",
+		DataFunc: fetchFalconDetectionDetailsData(loader),
+		Config:   makeDataSourceConfig(),
+		Args: &dataspec.RootSpec{
+			Attrs: []*dataspec.AttrSpec{
+				{
+					Name: "filter",
+					Type: cty.String,
+					Doc:  "Host search expression using Falcon Query Language (FQL)",
+				},
+				{
+					Name:        "size",
+					Type:        cty.Number,
+					Constraints: constraint.Integer | constraint.RequiredNonNull,
+					Doc:         "limit the number of queried items",
```

Comment:
Perhaps the size should have a default value
```suggestion
					Name:        "limit",
					Type:        cty.Number,
					Constraints: constraint.Integer | constraint.NonNull,
					Doc:         "Maximum number of returned items",
					DefaultVal: cty.NumberIntVal(50),
					MinInclusive: cty.NumberIntVal(1),
```
I guess that having a default limit would be a bit neater from the UX standpoint (@traut what do you think?)

Reply:
I agree. Let me update all prs

Reply:
I agree, let me update all prs